# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/arcticfox/hotfix-unaligned.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

SIGBUS unaligned memory access on SPARC

--- Arctic-Fox-46.1/image/decoders/nsIconDecoder.cpp.vanilla	2025-08-13 23:12:42.649123295 +0200
+++ Arctic-Fox-46.1/image/decoders/nsIconDecoder.cpp	2025-08-13 23:13:29.124477627 +0200
@@ -94,7 +94,8 @@
       return AsVariant(WriteState::NEED_MORE_DATA);  // Done with this row.
     }
 
-    uint32_t pixel = *reinterpret_cast<const uint32_t*>(aData);
+    uint32_t pixel;
+    memcpy(&pixel, aData, sizeof(pixel));
     aData += 4;
     aLength -= 4;
 
